doc/Iothub_sdk_options.md

```diff
@@ -0,0 +1,69 @@
+# IoTHub C SDK Options
+
+This document describes the options available in the c sdk and how to use the.
+
+- [Setting an Options](#set_option)
+- [IoThub Client Options](#IotHub_option)
+- [Transport Options](#transport_option)
+
+<a name="set_option"></a>
+
+## Setting an Option
+
+Setting and option in the c-sdk is dependant on which api set you are using.  If using the convience layer you would set the options in the api as follows:
+
+```c
+IOTHUB_CLIENT_RESULT IoTHubClient_SetOption(IOTHUB_CLIENT_HANDLE iotHubClientHandle, const char* optionName, const void* value)
+
+IOTHUB_CLIENT_RESULT IoTHubClient_LL_SetOption(IOTHUB_CLIENT_LL_HANDLE iotHubClientHandle, const char* optionName, const void* value)
+```
+
+An example of using a set option:
+
+```c
+bool trace_on = true;
+IoTHubClient_SetOption(sdk_handle, OPTION_LOG_TRACE, &trace_on);
+```
+
+## Available Option
+
+<a name="IotHub_option"></a>
+
+### IoThub_Client
+
+| Option Name        | Option Define              | Value Type         | Description                  |
+|--------------------|----------------------------|--------------------|-------------------------------|
+| `"messageTimeout"` | OPTION_MESSAGE_TIMEOUT     | tickcounter_ms_t*  | Timeout used for message      |
+| `"product_info"`   | OPTION_PRODUCT_INFO        | const char*        | Product information sent to the IoThub service on connection
```

Comment:
Can you add more details for the description?  Assuming someone has no idea what any of these features mean?

Reply:
I was hoping you would tell me what this means.

---
In reply to: [140865655](https://github.com/Azure/azure-iot-sdk-c/pull/241#discussion_r140865655) [](ancestors = 140865655)

Reply:
Haha :) @tameraw